src/crc16.rs

```diff
@@ -4,23 +4,22 @@ use core::hash::Hasher;
 use std::hash::Hasher;
 
 pub use util::make_table_crc16 as make_table;
+pub use util::CalcType;
 
 include!(concat!(env!("OUT_DIR"), "/crc16_constants.rs"));
 
-/// Structure that holds all of the important values for calculating a CRC
-///
-/// # Definitions
-///
-/// * **table:** Holds the table values based on the supplied polynomial for the fast CRC calculations
-/// * **initial:** The initial input value. AKA *reflect_in*
-/// * **value:** Holds the current value of the CRC
-/// * **reflect:** Chooses whether or not the CRC math is normal or reflected
-/// * **final_xor:** Final value to XOR with when calling Digest::sum16
+/// Structure that holds all of the important values for calculating a CRC.
+/// ### Details
+/// - **table**: Holds the table values based on the supplied polynomial for the fast CRC calculations
+/// - **initial**: The initial input value. AKA *reflect_in*
+/// - **value**: Holds the current value of the CRC
+/// - **reflect**: Chooses whether or not the CRC math is *Normal* or *Reverse*
+/// - **final_xor**: Final value to XOR with when calling `Digest::sum16()`. AKA *reflect_out*
 pub struct Digest {
     table: [u16; 256],
     initial: u16,
     value: u16,
-    reflect: bool,
+    reflect: CalcType,
```

Comment:
does it make sense to refactor out all reference to `reflect` or `rfl` since we're mostly using `calc` or `CalcType`?
if so, I can update them in a follow up pr

Reply:
btw thanks for putting this up

Reply:
"reflect" is the word that is used in other generic CRC algorithms.  You can change it if you want, but I'm not sure it makes a huge difference.